Checkstyle tests

nwmartin opened the pull request:
This completes the checkstyling of the tests. Of note, we've added a _temporary_ suppression rule that allows checkstyle to ignore public junit rules.

Commit:
Checkstylin' tests. ~180ish down on MockHTable.

Commit:
Down to 421 checkstyle errors.

Commit:
378 CheckStyle errors left.

Commit:
321 checkstyle errors remain.

Commit:
At 315. had to make a bunch of getters and spread them around. Fixed a test.

Commit:
Down to 181!

Commit:
Only 89 more...

Commit:
The checkstyle madness is over (for non-integrated tests, ie, most of them).

Commit:
Preliminary integration checkstyle.

Commit:
Merge branch 'master', remote-tracking branch 'origin'

Commit:
Merge branch 'master' into checkstyle-tests

Conflicts:
	src/test/java/com/softartisans/timberwolf/exchange/ExchangeMailStoreTest.java
	src/test/java/com/softartisans/timberwolf/exchange/ExchangeServiceTest.java
	src/test/java/com/softartisans/timberwolf/exchange/ExchangeTestBase.java
	src/test/java/com/softartisans/timberwolf/exchange/FindFolderTest.java
	src/test/java/com/softartisans/timberwolf/exchange/FindItemTest.java
	src/test/java/com/softartisans/timberwolf/exchange/GetItemTest.java
	src/test/java/com/softartisans/timberwolf/integrated/TestIntegration.java

Commit:
Checkstyling post-master merge. 316 left =/

Commit:
Checkstyle swapouts for private fields.

Commit:
Checkstyle count: 98.

Commit:
Fixed a test.

Commit:
Checkstyle...9 or so. Gotta refactor a function.

Commit:
Added comment suppressions of visibility modifier checks for junit rules using an @junitRule comment.

Commit:
Last of the checkstyling...again...for now.

Commit:
Merge remote-tracking branch 'origin/master'

ScottDugas commented on checkstyle.xml:
```diff
@@ -171,7 +181,7 @@
     <!-- Checks for common coding problems         -->
     <!-- See http://checkstyle.sf.net/config_coding.html -->
-    <module name="AvoidInlineConditionals"/>
+    <!-- <module name="AvoidInlineConditionals"/> Yeah, I don't think so CheckStyle. -->
```
Perhaps note that this means ternary statements

ScottDugas commented on src/test/java/com/softartisans/timberwolf/exchange/ExchangeEmailTest.java:
```diff
@@ -18,6 +19,9 @@
 import static org.mockito.Mockito.when;
 
+/**
+ * Test suite for the ExchangeEmail component.
```
ExchangeEmail class, component implies that it would be more than just the one class.

ScottDugas commented on src/test/java/com/softartisans/timberwolf/exchange/ExchangeMailStoreTest.java:
```diff
@@ -316,49 +322,60 @@ public void testGetMailWithPagingAndFolders() throws ServiceCallException, HttpE
+        ExchangeMailStore store = new ExchangeMailStore(getService(), findItemPageSize, getItemPageSize);
         Iterator<MailboxItem> mail = store.getMail(defaultUsers).iterator();
-        for (String folder : new String[] { "FOLDER-ONE-ID", "FOLDER-TWO-ID", "FOLDER-THREE-ID" })
+        final int folderIdTwoCount = 13;
```
rename: folderTwoIdCount

ScottDugas commented on src/test/java/com/softartisans/timberwolf/exchange/ExchangeMailStoreTest.java:
```diff
@@ -316,49 +322,60 @@ public void testGetMailWithPagingAndFolders() throws ServiceCallException, HttpE
         Iterator<MailboxItem> mail = store.getMail(defaultUsers).iterator();
-        for (String folder : new String[] { "FOLDER-ONE-ID", "FOLDER-TWO-ID", "FOLDER-THREE-ID" })
+        final int folderIdTwoCount = 13;
+        final int folderIdOtherCount = 2;
```
rename: otherFolderIdCount

ScottDugas commented on src/test/java/com/softartisans/timberwolf/exchange/ExchangeTestBase.java:
```diff
@@ -39,17 +45,45 @@
     /** This is the name of our default folder. */
-    protected String defaultFolderId = "ANAMAZINGLYENGLISH-LIKEGUID";
-    protected final String defaultUser = "bkerr";
+    private String defaultFolderId = "ANAMAZINGLYENGLISH-LIKEGUID";
```
final?

ScottDugas commented on src/test/java/com/softartisans/timberwolf/exchange/ExchangeServiceTest.java:
```diff
@@ -29,125 +28,128 @@
 import static org.mockito.Mockito.when;
 
+/**
+ * Test suite for the ExchangeService component.
```
This could also be class, but whatever

ScottDugas commented on src/test/java/com/softartisans/timberwolf/integrated/HTableResource.java:
```diff
@@ -46,8 +47,10 @@ public HTableResource()
     public Statement apply(final Statement inner, final Description description)
     {
+        final int arbitraryValue1 = 130;
+        final int arbitraryValue2 = 32;
```
These are my favorite new variables to replace magic numbers

ScottDugas commented on src/test/java/com/softartisans/timberwolf/integrated/TestHBaseIntegration.java:
```diff
@@ -10,15 +13,17 @@
+ */
 public class TestHBaseIntegration
 {
+    // @junitRule
```
Should we comment when we do this, at least to look at checkstyle, or rename it: "checkstyle: @Rule field can be public"
It's a bit more wordy, but it will be clear when someone's looking at this code, and hasn't looked at the checkstyle config (because why would they). Or should we comment on this in the README under the coding conventions, or running the mvn checkstyle.

ScottDugas commented:
basically here are the things that needed to be fixed:
- reorder imports
- whitespace
- start second line with operator when wrapping
- NO constant numbers
- new checkstyle filter so that JUnit @Rules can be public fields
- checkstyle allows ternary operators
- new javadocs
- end javadoc with period
- final parameters
- must be private fields with protected getters no protected fields
- casing on constants
- a handful of methods with Pascal casing
- integration test was over 150 lines

Commit:
Checkstyle review changes.

Commit:
Quick master merge before merging.
Merge branch 'master' into checkstyle-tests

Conflicts:
	src/test/java/com/softartisans/timberwolf/exchange/ExchangeMailStoreTest.java
